src/lib/syncable_ledger/syncable_ledger.ml

```diff
@@ -340,7 +399,47 @@ end = struct
             Either.First
               (Num_accounts
                  (len, MT.get_inner_hash_at_addr_exn mt content_root_addr) )
+        | What_child_hashes (a, subtree_depth) -> (
+            match subtree_depth with
+            | n
+              when n >= 1
+                   && n <= Context.compile_config.sync_ledger_max_subtree_depth
```

Comment:
This isn't a protocol violation. We should just take the `min` of the configured max and `n`.